### PR TITLE
(test) Move test_runtime to ghcr_test_runtime; adapt workflows

### PR DESCRIPTION
**Pull request by tobitege**
**What is the problem that this fixes or functionality that this introduces? Does it fix any open issues?**

- Solves #3504 : move/rename old `test_runtime.py` out of `tests/unit` folder to `tests/ghcr_test_runtime.py` 
- Adapt related workflows (remove reference from unit tests; rename in ghcr_runtime.yml)
- Fix a runtime AttributeError in `container_image` about missing `param` (e.g. if run in WSL):
    ```sh
    ERROR tests/unit/test_runtime.py::test_bash_python_version[EventStreamRuntime] - AttributeError: 'SubRequest' object has no attribute 'param'
  ERROR tests/unit/test_runtime.py::test_nodejs_22_version[EventStreamRuntime] - AttributeError: 'SubRequest' object has no attribute 'param'
  ERROR tests/unit/test_runtime.py::test_go_version[EventStreamRuntime] - AttributeError: 'SubRequest' object has no attribute 'param'
```

---
**Give a summary of what the PR does, explaining any non-trivial design decisions**

This PR is to avoid confusion about a test that was formerly run as a regular unit test,
but is now part of a special workflow run.

### CODE REVIEW

File: .github/workflows/ghcr_runtime.yml [hunk 1]
```diff
@@ -191,7 +191,7 @@ jobs:
           # Print the full name of the image
           echo "Loaded Docker image: $image_name"
 
-          TEST_RUNTIME=${{ matrix.runtime_type }} SANDBOX_USER_ID=$(id -u) SANDBOX_CONTAINER_IMAGE=$image_name TEST_IN_CI=true poetry run pytest --cov=agenthub --cov=openhands --cov-report=xml -s ./tests/unit/test_runtime.py
+          TEST_RUNTIME=${{ matrix.runtime_type }} SANDBOX_USER_ID=$(id -u) SANDBOX_CONTAINER_IMAGE=$image_name TEST_IN_CI=true poetry run pytest --cov=agenthub --cov=openhands --cov-report=xml -s ./tests/ghcr_test_runtime.py
```
Comment (xingyaoww) [inline .github/workflows/ghcr_runtime.yml:194]: Can we actually rename to `tests/runtime` folder? I was actually thinking we can break one really long `test_runtime.py` file into multiple smaller files, like `test_bash_cmd.py`, `test_ipython.py`, `test_browsing.py`, etc?

File: tests/runtime/test_bash.py [hunk 2]
```diff
@@ -0,0 +1,744 @@
...
+            request.param = request.config.getoption('--container-image')
+        if request.param is None:
+            request.param = pytest.param(
+                'nikolaik/python-nodejs:python3.11-nodejs22',
+                'python:3.11-bookworm',
+                'node:22-bookworm',
+                'golang:1.23-bookworm',
+            )
+    print(f'Container image: {request.param}')
+    return request.param
```
Comment (xingyaoww) [inline tests/runtime/test_bash.py:89]: Should we put these into a separate contest file?https://docs.pytest.org/en/stable/reference/fixtures.html#conftest-py-sharing-fixtures-across-multiple-files

Comment (tobitege) [reply tests/runtime/test_bash.py:89]: Makes total sense, will look into it

**Review by xingyaoww — APPROVED** [review]
LGTM -- Thanks!

**Pull request merged**